src/modi/c13.rs

```diff
@@ -583,8 +589,270 @@ impl<'a> DebugFileChecksumsSubsection<'a> {
+    fn parse(data: &'a [u8]) -> Result<Self> {
+        Ok(Self { data })
+    }
+
+    fn imports(self) -> CrossScopeImportModuleIter<'a> {
+        let buf = ParseBuffer::from(self.data);
+        CrossScopeImportModuleIter { buf }
+    }
+}
+
+/// Provides efficient access to imported types and IDs from other modules.
+///
+/// This can be used to resolve cross module references. See [`ItemIndex::is_cross_module`] for more
+/// information.
+#[derive(Clone, Debug, Default)]
+pub struct CrossModuleImports<'a> {
+    modules: Vec<CrossScopeImportModule<'a>>,
+}
+
+impl<'a> CrossModuleImports<'a> {
+    pub(crate) fn parse(data: &'a [u8]) -> Result<Self> {
+        let export_data = DebugSubsectionIterator::new(data)
+            .find(|sec| sec.kind == DebugSubsectionKind::CrossScopeImports)?
+            .map(|sec| sec.data);
+
+        let section = match export_data {
+            Some(d) => DebugCrossScopeImportsSubsection::parse(d)?,
+            None => return Ok(Self::default()),
+        };
+
+        let modules = section.imports().collect()?;
+        Ok(Self { modules })
+    }
+
+    /// Resolves the referenced module and local index for the index.
+    ///
+    /// The given index **must** be a cross module reference. Use `ItemIndex::is_cross_module` to
+    /// check this before invoking this function. If successful, this function returns a reference
+    /// to the module that declares the type, as well as the local index of the type in that module.
+    ///
+    /// # Errors
+    ///
+    /// * `Error::NotACrossModuleRef` if the given index is already a global index and not a cross
+    ///   module reference.
+    /// * `Error::CrossModuleRefNotFound` if the cross module reference points to a module or local
+    ///   index that is not indexed by this import table.
+    pub fn resolve_import<I>(&self, index: I) -> Result<CrossModuleRef<I>>
+    where
+        I: ItemIndex,
+    {
+        let raw_index = index.into();
+        if !index.is_cross_module() {
+            return Err(Error::NotACrossModuleRef(raw_index));
+        }
+
+        let module_index = (raw_index & 0x7ff0_0000) as usize;
+        let import_index = (raw_index & 0x000f_ffff) as usize;
+
+        let module = self
+            .modules
+            .get(module_index)
+            .ok_or_else(|| Error::CrossModuleRefNotFound(raw_index))?;
+
+        let local_index = module
+            .get(import_index)
+            .ok_or_else(|| Error::CrossModuleRefNotFound(raw_index))?;
+
+        Ok(CrossModuleRef(module.name, local_index))
+    }
+}
+
+/// Raw representation of `CrossModuleExport`.
+///
+/// This type can directly be mapped onto a slice of binary data and exposes the underlying `local`
+/// and `global` fields with correct endianness via getter methods. There are two ways to use this:
+///
+///  1. Binary search over a slice of exports to find the one matching a given local index
+///  2. Enumerate all for debugging purposes
+#[repr(C, packed)]
```

Comment:
[Nomicon frowns](https://doc.rust-lang.org/nomicon/other-reprs.html#reprpacked):

> As of Rust 2018, this still can cause undefined behavior.
> 
> `repr(packed)` is not to be used lightly. Unless you have extreme requirements, this should not be used.

More generally, could we implement this in safe Rust without a performance penalty? [`zerocopy`](https://crates.io/crates/zerocopy) comes to mind.

Reply:
Fair point. In this case we could even remove the `packed` since we can assume (or validate) data alignment in the PDB.

Adding `zerocopy` for this one case is probably an overkill, mostly because it adds yet another way of deserializing in addition to `scroll` and `ParseBuf`. Your call.

Reply:
I'm realizing that I've introduced most of the `repr(packed)` that we currently have here. Most importantly OMAPs, which function the same way as cross module exports.